### PR TITLE
Fork1 tests v2

### PR DESCRIPTION
**Pull request by ercecan**
Tests these => https://github.com/chainwayxyz/citrea/issues/1464#issuecomment-2537022638

### CODE REVIEW

File: crates/evm/src/evm/test_data/KZGPointEvaluation.sol
Comment (eyusufatik) [inline crates/evm/src/evm/test_data/KZGPointEvaluation.sol:1]: rename this to something like KZGPointEvaluationCaller

Comment (eyusufatik) [reply crates/evm/src/evm/test_data/KZGPointEvaluation.sol:1]: and also other related structs.

File: crates/evm/src/tests/fork_tests.rs [hunk 1]
```diff
@@ -535,3 +556,239 @@ fn test_blob_base_fee_should_return_1() {
...
+        da_slot_txs_commitment: [42u8; 32],
+        pre_state_root: [10u8; 32].to_vec(),
+        current_spec: SovSpecId::Fork1,
+        pub_key: vec![],
+        deposit_data: vec![],
+        l1_fee_rate,
+        timestamp: 0,
+    };
+
+    // Deploy transient storage contract
```
Comment (eyusufatik) [inline crates/evm/src/tests/fork_tests.rs]: fix comment

File: crates/evm/src/tests/fork_tests.rs [hunk 2]
```diff
@@ -535,3 +556,239 @@ fn test_blob_base_fee_should_return_1() {
...
+        da_slot_txs_commitment: [42u8; 32],
+        pre_state_root: [10u8; 32].to_vec(),
+        current_spec: SovSpecId::Genesis,
+        pub_key: vec![],
+        deposit_data: vec![],
+        l1_fee_rate,
+        timestamp: 0,
+    };
+
+    // Deploy transient storage contract
```
Comment (eyusufatik) [inline crates/evm/src/tests/fork_tests.rs]: comment

File: crates/evm/src/tests/fork_tests.rs [hunk 3]
```diff
@@ -535,3 +556,239 @@ fn test_blob_base_fee_should_return_1() {
...
+    evm.finalize_hook(&[99u8; 32].into(), &mut working_set.accessory_state());
+    sleep(std::time::Duration::from_secs(2));
+    l2_height += 1;
+
+    let offchain_code = evm
+        .offchain_code
+        .get(&code_hash, &mut working_set.offchain_state());
+
+    assert!(offchain_code.is_none());
+
```
Comment (eyusufatik) [inline crates/evm/src/tests/fork_tests.rs:754]: make tx on the contract that was deployed before fork1 and see that you can read it from offchain storage afterwards

**Review by eyusufatik — CHANGES_REQUESTED** [review]

**Review by rakanalh — APPROVED** [review]

**Comment by eyusufatik** [comment]
holding on to this pr until we get our upgrade prs merged. because disabling point evaluation precompile needs some changes unfortunately.

File: crates/evm/src/evm/handler.rs [hunk 4]
```diff
@@ -314,6 +314,29 @@ struct CitreaHandler<SPEC, EXT, DB> {
...
+
+            pub fn kzg(_input: &Bytes, _gas_limit: u64, _env: &Env) -> PrecompileResult {
+                pub const GAS_COST: u64 = 0;
+                pub const RETURN_VALUE: &[u8; 64] = &[0; 64];
+                Ok(PrecompileOutput::new(GAS_COST, RETURN_VALUE.into()))
+            }
+
+            let mut precompiles = revm::handler::mainnet::load_precompiles::<SPEC, DB>();
+            let precompiles_inner = precompiles.to_mut();
+            if let Some(kzg_point_evaluation) = precompiles_inner.get_mut(&u64_to_address(0x0A)) {
```
Comment (eyusufatik) [inline crates/evm/src/evm/handler.rs]: since we know at what point kzg comes into play, we can remove only then with the index :)

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/evm/src/evm/handler.rs (hunk 4 above)
Comment (eyusufatik) [reply crates/evm/src/evm/handler.rs]: also the costs should not be zero. it should be the same as call gas cost to a regular address with no precompile

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**